packages/dds/tree/src/simple-tree/schemaFactory.ts

```diff
@@ -131,17 +131,27 @@ type UnbrandedName<T extends FlexLeafNodeSchema> = T["name"] extends TreeNodeSch
  *
  * @sealed @public
  */
-export class SchemaFactory<TScope extends string = string, TName extends number | string = string> {
+export class SchemaFactory<
+	TScope extends string | undefined = string,
+	TName extends number | string = string,
+> {
 	private readonly structuralTypes: Map<string, TreeNodeSchema> = new Map();
 
 	/**
 	 * @param scope - Prefix appended to the identifiers of all {@link TreeNodeSchema} produced by this builder.
 	 * Use of [Reverse domain name notation](https://en.wikipedia.org/wiki/Reverse_domain_name_notation) or a UUIDv4 is recommended to avoid collisions.
+	 * You may opt out of using a scope by passing `undefined`, but note that this increases the risk of collisions.
 	 */
 	public constructor(public readonly scope: TScope) {}
 
-	private scoped<Name extends TName | string>(name: Name): `${TScope}.${Name}` {
-		return `${this.scope}.${name}`;
+	// TODO: is `| string` needed here?
+	private scoped<Name extends TName | string>(
+		name: Name,
+	): TScope extends undefined ? `${Name}` : `${TScope}.${Name}` {
+		// TODO: is this cast really necessary?
```

Comment:
Rewriting the and runtime to like the second option below) might fix it.

Also factoring the type logic out to a help now that its more complicated may be a good idea. There are a lot of copies of this logic through the API, and a helper may be more readable and more maintainable (It will have to be public  package exported, which is annoying, but that also gives you a place to document this new policy, and to link when talking about scoped names)

```typescript
type ScopedName<
	TScope extends undefined | string,
	Name extends number | string,
> = TScope extends undefined ? `${Name}` : `${TScope}.${Name}`;

type ScopedName<
	TScope extends undefined | string,
	Name extends number | string,
> = `${TScope}${TScope extends undefined ? "" : "."}${Name}`;
```

Reply:
That still didn't satisfy TSC, unfortunately. And I think I find the original format a bit easier to parse, so I think I'll stick with what I had.